packages/cardano-services/README.md

```diff
@@ -1,7 +1,92 @@
+
+## Features
+- [CLI] or run scripts for the [HTTP server](./src/run.ts) and [worker](./src/startWorker.ts) with 
+configuration via environment variables and _optional_ loading of secrets from disk.
+- Service port discovery via DNS resolution, or static configuration.
+- Fault-tolerant transaction submission via persistent queue, or direct submission.
+- _Optional_ [Prometheus] metrics available at `/metrics`
+
+## Services
+### HTTP Server
+The HTTP server can be started with one or more provider modules by name, segmented by URL path. 
+Run the [CLI] with `start-server --help` to see the full list of options, or inspect the 
+environment variables within [./src/run.ts](./src/run.ts). 
+
+### Worker
+A worker must be started when opting for queue-based transaction submission.
+Run the [CLI] with `start-worker --help` to see the full list of options, or inspect the
+environment variables within [./src/startWorker.ts](./src/startWorker.ts).
+
+
+## Examples
+
+_The following examples require the [install and build] steps 
+to be completed._
+
+#### All Providers | Static Service Config | Direct Tx Submission
+
+- The server will expose all [Provider] HTTP services
+- Transactions will be submitted directly via [Ogmios], running at `ws://localhost:1338`.
+- Connects to [PostgreSQL] service running at `localhost:5432`
+- HTTP API exposed using a custom API URL
+
+``` console
+./dist/cli.js \
+  start-server \
+    --api-url http://localhost:6000 \
+    --cardano-node-config-path ./config/cardano-node/config.json \
+    --ogmios-url  ws://localhost:1338 \
+    --db-connection-string postgresql://somePgUser:somePassword@localhost:5432/someDbName \
+    asset,chain-history,stake-pool,tx-submit,network-info,utxo
+```
+
+#### All Providers | Service Discovery | Queued Tx Submission | Metrics
+
+- The server will expose all [Provider] HTTP services
+- Transactions will be queued by the HTTP service, with the worker completing the
+  submission. The HTTP service receives the submission result via a dedicated channel to
+  complete the HTTP request.
+- Ports for [Ogmios], [PostgreSQL], and [RabbitMQ], discovered using DNS resolution.
+- HTTP API exposed using a custom API URL
+- Prometheus metrics exporter enabled at http://localhost:6000/metrics
+
+``` console
+./dist/cli.js \
+  start-server \
+    --api-url http://localhost:6000 \
+    --cardano-node-config-path ./config/cardano-node/config.json \
+    --enable-metrics true \
+    --ogmios-srv-service-name  some-domain-for-ogmios \
+    --postgres-db someDbName \
+    --postgres-password somePassword \
+    --postgres-srv-service-name \
+    --postgres-user somePgUser \
+    --rabbitmq-srv-service-name some-domain-for-rabbitmq \
+    --use-queue \
+    asset,chain-history,stake-pool,tx-submit,network-info,utxo
+```
+``` console
+./dist/cli.js \
+  start-worker \
+    --ogmios-srv-service-name  some-domain-for-ogmios \
+    --rabbitmq-srv-service-name some-domain-for-rabbitmq
+```
 
```

Comment:
Some visual representation of interaction between all components for both deployment profiles would go a long way. You can use https://github.blog/2022-02-14-include-diagrams-markdown-files-mermaid/ to do this on GitHub.

Reply:
I've purposely omitted architecture diagrams, as it's being worked on in the lower application context, which we intend to hoist once complete.

Reply:
Well, this README is targeted at GitHub readers, I assume. It can link to better more in-depth architecture description, but it could still contain some pictures to get a quick understanding of what's going on.

Reply:
We'll consider this in the next round of improvements once the other documentation has been considered for adoption.